File: q1/q1.js
var line;
var conv_dict={
    "thou":1, "th":1,
    "inch":        1000, "in":       1000,
    "foot":       12000, "ft":      12000,
    "yard":       36000, "yd":      36000,
    "chain":     792000, "ch":     792000,
    "furlong":  7920000, "fur":   7920000,
    "mile":    63360000, "mi":   63360000,
    "league": 190080000, "lea": 190080000
};



function compute(line){
    var s=line.split(' ');
    var u=s[0];
    var f=s[1];
    var t=s[3];
    
    return u * conv_dict[f]/conv_dict[t];
    
}
while (line = readline()) {
    
    print(compute(line));
}

File: q1/q1.cs
using System;

public class Test
{
    public static void Main()
    {
        // your code goes here
        int[] intArr = new int[] {3,8,10,22,3,12,1000};
        int startIndex = 0, endIndex = 0;
        
        string line = Console.ReadLine();
        string[] words = line.Split(' ');

        switch(words[1]){
                
                    case "thou": 
                    case "th"  : startIndex = 7;break;
                        
                    case "inch":
                    case "in": startIndex = 6;break;
                    
                    case "foot":
                    case "ft":startIndex = 5;break;
                        
                    case "yard": 
                    case "yd"  : startIndex = 4; break;
                        
                    case "chain":
                    case "ch": startIndex = 3;break;
                    
                    case "furlong":
                    case "fur": startIndex = 2;break;
                        
                    case "mile": 
                    case "mi"  : startIndex = 1; break;
                        
                    case "league":
                    case "lea": startIndex = 0; break;
                    
                    
                    
                
                    default: break;
                }
                
                switch(words[3]){
                
                case "thou": 
                case "th"  : endIndex = 7;break;
                    
                case "inch":
                case "in": endIndex = 6;break;
                
                case "foot":
                case "ft":endIndex = 5;break;
                    
                case "yard": 
                case "yd"  : endIndex = 4; break;
                    
                case "chain":
                case "ch": endIndex = 3;break;
                
                case "furlong":
                case "fur": endIndex = 2;break;
                    
                case "mile": 
                case "mi"  : endIndex = 1; break;
                    
                case "league":
                case "lea": endIndex = 0; break;

            
                default: break;
            }
            
            double result = double.Parse(words[0]);
            if (startIndex < endIndex){
                //multiply all
                for (int i = startIndex; i < endIndex; i++){
                    result *= intArr[i];
                }
            }
            if (startIndex > endIndex){
                //divide all
                for (int i = startIndex-1; i >= endIndex; i--){
                    result /= intArr[i];
                }
            }
            Console.WriteLine(result);
    }
}
